.github/pull_request_template.md

```diff
@@ -0,0 +1,41 @@
+## Describe your changes
+< Summary of the changes.>
+
+< Please also include relevant motivation and context. >
+
+< List any dependencies that are required for this change. >
+
+## Type of change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist before requesting a review
+
+- [ ] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
```

Comment:
I completely agree with the added README suggestion, but I am not entirely convinced we should discourage the use of inline comments. Maybe the text could be changed to something like 
"I have commented my code where necessary, for example in hard-to-understand areas"

Reply:
How about we write:

```markdown
- [ ] I have added docstrings to any new/modified functions
- [ ] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the [README](README.MD) to cover introduced code changes
```

Reply:
I think that is good. Test (linting) should catch missing docstrings right? So that is maybe superfluous if test pass?

Reply:
> Test (linting) should catch missing docstrings right? So that is maybe superfluous if tests pass?

True, the linting checks that docstrings have been written, but not that they are useful. Maybe we should leave this in to just remind people to think through the docstrings?

Reply:
Yes, I think that's good. Proposal:
```suggestion
- [ ] For any new/modified functions/classes I have added docstrings that clearly describe its purpose, expected inputs and returned values
- [ ] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the [README](README.MD) to cover introduced code changes
```

Reply:
Update the text about doc-strings